pkg/util/export/batch_processor.go

```diff
@@ -90,32 +90,28 @@ func (b *bufferHolder) Add(item batchpipe.HasName) {
 	}
 }
 
-// Stop set bufferHolder readonly, and hold Add request
-func (b *bufferHolder) Stop() bool {
+// StopAndGetBatch set bufferHolder readonly, which can hold Add request,
+// and gen batch request content
+func (b *bufferHolder) StopAndGetBatch(buf *bytes.Buffer) bool {
 	b.mux.RLock()
 	defer b.mux.RUnlock()
 	if !atomic.CompareAndSwapUint32(&b.readonly, READWRITE, READONLY) {
```

Comment:
So b.mux will protect the data structure?    In that case do we really need atomic?

Reply:
seems a little bit redundant, i will reconsider it.

Reply:
fixed, just use `RWMutex` as protection.